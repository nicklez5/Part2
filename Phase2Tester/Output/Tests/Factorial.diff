--- SelfTestCases/Factorial.out	2011-04-28 14:55:20.000000000 -0700
+++ ./Output/Tests/Factorial.out	2020-02-16 14:52:29.168188785 -0800
@@ -1 +1,4 @@
-3628800
+--- Error (ProgramFault) ---
+expecting argument 2 to be an integer, found undefined value
+  Main() line 12
+  Fac.ComputeFac(<Heap.0>+0 10) line 23
