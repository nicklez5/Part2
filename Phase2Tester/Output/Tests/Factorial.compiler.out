   const vmt_Fac
      :Fac.ComputeFac

   func Main()
      t.0 = HeapAllocZ(4)
      [t.0] = :vmt_Fac
      if t.0 goto :null0
      Error("null pointer")
      null0:
      t.1 = [t.0]
      t.1 = [t.1+0]
      t.2 = call t.1(t.0 10)
      PrintIntS(t.2)
      ret

   func Fac.ComputeFac(this num)
      t.2 = LtS(num 1)
      t.2 = Eq(t.2 1)
      if0 t.2 goto :if1_else
         num_aux = 1
         goto :if1_end
      if1_else:
         t.3 = MulS(num null)
         num_aux = t.3
         if1_end:
         ret num_aux

